app/forms/declaration/RepresentativeDetails.scala

```diff
@@ -40,8 +40,10 @@ object RepresentativeDetails extends DeclarationPage {
         text().verifying("supplementary.representative.representationType.error.wrongValue", isContainedIn(representativeStatusCodeAllowedValues))
       )
     )(RepresentativeDetails.apply)(RepresentativeDetails.unapply)
-    .verifying("supplementary.namedEntityDetails.error", details => details.details.nonEmpty || details.statusCode.isEmpty)
-    .verifying("supplementary.representative.representationType.error.empty", details => details.details.isEmpty || details.statusCode.nonEmpty)
+    .verifying(
+      "supplementary.representative-details.representative-details-page.error",
+      details => (details.statusCode.isEmpty && details.details.map(_.eori).isEmpty) || details.statusCode.nonEmpty
```

Comment:
Not 100% sure but do we not need to check that EORI is non-empty here?  So either (statusCode empty and eori empty) or (statusCode non-empty and eori non-empty).

Actually, forget that.  We need a way of NOT setting the representative details and once the user has picked a statusCode there's no way to "un-pick" it (i.e. clear the radio option).

So if that logic is true, then do we need this extra 'verifying' step at all?

Reply:
The user can leave the page blank only if no EORI is supplied. If he selects one radio and then wants to clear, there is no clean way to do it. I discussed this with Patryk yesterday. It seems to be a recurring pattern in some pages. In the future, we might have a ticket to add a "no option" radio to this kind of page.

Reply:
Thought so,  thanks for confirming